core.impl/src/main/java/com/ericsson/bss/cassandra/ecchronos/core/impl/locks/HostStatesImpl.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2024 Telefonaktiebolaget LM Ericsson
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ericsson.bss.cassandra.ecchronos.core.impl.locks;
+
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.metadata.Metadata;
+import com.ericsson.bss.cassandra.ecchronos.core.impl.logging.ThrottlingLogger;
+import com.ericsson.bss.cassandra.ecchronos.core.jmx.DistributedJmxProxy;
+import com.ericsson.bss.cassandra.ecchronos.core.jmx.DistributedJmxProxyFactory;
+import com.ericsson.bss.cassandra.ecchronos.core.locks.HostStates;
+import com.ericsson.bss.cassandra.ecchronos.core.metadata.DriverNode;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import com.datastax.oss.driver.api.core.metadata.Node;
+import com.google.common.annotations.VisibleForTesting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of the {@link HostStates} interface using JMX to retrieve node statuses and then caches the retrieved
+ * statuses for some time.
+ */
+public final class HostStatesImpl implements HostStates, Closeable
```

Comment:
This class already exists in com.ericsson.bss.cassandra.ecchronos.core.impl.repair.state.HostStatesImpl

Reply:
I cannot find this. there is one class ReplicationStateImpl in that package.

Reply:
after rebase found and replaced